plutus-tutorial/tutorial/Tutorial/03-wallet-api.md

```diff
@@ -99,30 +100,22 @@ In the crowdfunding campaign the data script contains a `Contributor` value, whi
 
 ## 1.2 The Validator Script
 
-The general form of a validator script is `Redeemer -> DataScript -> PendingTx -> Answer`. That is, the validator script is a function of three arguments that produces a value of type `Answer` (or fails with an error). As contract authors we can freely choose the types of `Redeemer`, `DataScript` and `Answer`. The third argument has to be of type `PendingTx` because that is the information about the current transaction, provided by the slot leader. When the evaluation of the script has finished without an error, the result is discarded and never used again. The value of `Answer` is not relevant, and therefore we usually choose `Answer` to be the unit type `()`.
-
-Since we already know the types of data and redeemer scripts (`Contributor` and `CampaignAction`, respectively), we automatically know the signature of the validator script:
+The general form of a validator script is `RedeemerScript -> DataScript -> PendingTx -> Answer`. The types of data and redeemer scripts are `Contributor` and `CampaignRedeemer`, respectively, so the signature of the validator script is:
 
 ```haskell
-type CampaignValidator = CampaignAction -> Contributor -> PendingTx -> ()
+type CampaignValidator = CampaignRedeemer -> Contributor -> PendingTx -> ()
 ```
 
-`CampaignValidator` is a function that takes three parameters - `CampaignAction`, `Contributor`, and `PendingTx` - and produces a unit value `()` or fails with an error.
-
-If we want to implement `CampaignValidator` we need to know the parameters of the campaign, so that we can check if the selected `CampaignAction` is allowed. In Haskell we can do this by writing a function `mkValidator :: Campaign -> CampaignValidator` that takes a `Campaign` and produces a `CampaignValidator`. However, we need to wrap `mkValidator` in Template Haskell quotes so that it can be compiled to Plutus Core. We therefore write `mkValidator` like this:
+If we want to implement `CampaignValidator` we need to have access to the parameters of the campaign, so that we can check if the selected `CampaignAction` is allowed. In Haskell we can do this by writing a function `mkValidator :: Campaign -> CampaignValidator` that takes a `Campaign` and produces a `CampaignValidator`. However, we need to wrap `mkValidator` in Template Haskell quotes so that it can be compiled to Plutus Core. We therefore write `mkValidator` like this:
```

Comment:
You need to say something about the way we pass parameters to scripts as arguments before you do it below.